plugin/pkg/skplug/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,128 @@
+package dispatcher
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-plugin"
+	"github.com/josephburnett/sk-plugin/pkg/skplug"
+	"github.com/josephburnett/sk-plugin/pkg/skplug/proto"
+	"os"
+	"os/exec"
+	"sync"
+)
+
+type Dispatcher interface {
+	Init(pluginsPaths []string)
+	Shutdown()
+	Event(partition string, time int64, typ proto.EventType, object skplug.Object) error
+	Stat(partition string, stat []*proto.Stat) error
+	HorizontalRecommendation(partition string, time int64) (rec int32, err error)
+	VerticalRecommendation(partition string, time int64) (rec []*proto.RecommendedPodResources, err error)
+}
+
+type dispatcher struct {
```

Comment:
Please add type assertions which tell us what interfaces this satisfied, and enforces them at compile time (rather than runtime).

E.g.
```
var _ Dispatcher = dispatcher{}
var _ Plugin = dispatcher{}
```

Of course, if you want to assert dispatcher is also a plugin, you will have to either implement the `Capabilities` API (which kind of makes sense) or split the `Plugin` interface into two parts: https://github.com/pivotal/skenario/blob/d84bd1354d1dc8908a12fea9737f792ef86cc5c8/plugin/pkg/skplug/interface.go#L25-L28

Reply:
Did some changes. Please review to what extends my changes suits to what is said here.

Reply:
Yes yes.  This looks much more clear.  The `Dispatcher` interface describes the functionality for being a dispatcher.  `Plugin` that for being a plugin.  And you say explicitly that this dispatcher implementation is both.  Nice.